src/ProjectOrigin.WalletSystem.IntegrationTests/FlowTests/ClaimTests.cs

```diff
@@ -0,0 +1,98 @@
+using ProjectOrigin.WalletSystem.IntegrationTests.TestClassFixtures;
+using ProjectOrigin.WalletSystem.Server;
+using System;
+using System.Threading.Tasks;
+using ProjectOrigin.WalletSystem.V1;
+using Xunit;
+using ProjectOrigin.WalletSystem.Server.Models;
+using Xunit.Abstractions;
+using ProjectOrigin.PedersenCommitment;
+using ProjectOrigin.WalletSystem.IntegrationTests.TestExtensions;
+using ProjectOrigin.Common.V1;
+using Npgsql;
+using Dapper;
+using FluentAssertions;
+
+namespace ProjectOrigin.WalletSystem.IntegrationTests.FlowTests;
+
+public class ClaimTests : WalletSystemTestsBase, IClassFixture<RegistryFixture>, IClassFixture<InMemoryFixture>
+{
+    private readonly RegistryFixture _registryFixture;
+
+    public ClaimTests(
+            GrpcTestFixture<Startup> grpcFixture,
+            PostgresDatabaseFixture dbFixture,
+            InMemoryFixture inMemoryFixture,
+            RegistryFixture registryFixture,
+            ITestOutputHelper outputHelper)
+            : base(
+                  grpcFixture,
+                  dbFixture,
+                  inMemoryFixture,
+                  outputHelper,
+                  registryFixture)
+    {
+        _registryFixture = registryFixture;
+    }
+
+    private async Task<FederatedStreamId> IssueCertToDepositEndpoint(DepositEndpoint senderDepositEndpoint, uint issuedAmount, Electricity.V1.GranularCertificateType type)
+    {
+        var prodCommitment = new SecretCommitmentInfo(issuedAmount);
+        var position = 1;
+        var issuedEvent = await _registryFixture.IssueCertificate(type, prodCommitment, senderDepositEndpoint.PublicKey.Derive(position).GetPublicKey());
+        await _dbFixture.InsertSlice(senderDepositEndpoint, position, issuedEvent, prodCommitment);
+        return issuedEvent.CertificateId;
+    }
+
+    [Fact]
+    public async Task Claim150_FromTwoLargerSlices_Success()
```

Comment:
Why only claim 150 when consumption is 300?

Reply:
To test both slices are sliced and then claimed the requested 150, the wallet allows partial claiming, it doesn't force claiming everything one has.